qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -287,29 +288,36 @@ def job_limit(self) -> BackendJobLimit:
                 maximum_jobs = backend_job_limit.maximum_jobs
                 running_jobs = backend_job_limit.running_jobs
 
-            * If ``maximum_jobs`` or ``running_jobs`` are ``None``, the
-                job limit information is currently not available.
-            * If ``maximum_jobs`` is equal to ``-1``, then there are
+            * If ``maximum_jobs`` is equal to ``None``, then there are
                 no limits to the maximum number of concurrent jobs a user
                 could submit to the backend at a time.
 
         Returns:
             the job limit for the backend with this provider.
 
         Raises:
-            LookupError: If jobs limit for the backend can't be found.
+            IBMQBackendApiProtocolError: If an unexpected value received from the server.
+            IBMQBackendJobLimitDataNotAvailable: If the job limit information is not available.
         """
         api_job_limit = self._api.backend_job_limit(self.name())
 
         try:
-            return BackendJobLimit.from_dict(api_job_limit)
+            job_limit = BackendJobLimit.from_dict(api_job_limit)
+            if job_limit.maximum_jobs is None or job_limit.running_jobs is None:
+                raise IBMQBackendJobLimitDataNotAvailable(
+                    'Job limit data for the backend is not available.')
+            if job_limit.maximum_jobs == -1:
+                # Manually set `maximum` to `None` if backend has no limits.
+                job_limit.maximum_jobs = None
+            return job_limit
         except ValidationError as ex:
-            raise LookupError(
-                "Couldn't get backend jobs limit: {0}".format(ex))
+            raise IBMQBackendApiProtocolError(
+                'Unexpected return value from the server when '
+                'querying job limit data for the backend: {}.'.format(ex))
 
     def remaining_jobs_count(self) -> Optional[int]:
         """Return the number of remaining jobs that could be submitted to the backend.
-
+F
```

Comment:
I don't think you meant to add this lol

Reply:
I can't remember but I'm sure there is a perfectly good explanation for it... 😅. Thank goodness for Git!